draft-ietf-httpbis-cache-digest.md

```diff
@@ -182,64 +170,171 @@ response was fresh or stale.
+
+1. Let `f` be the number of bits per fingerprint, calculated as `P + 3`
+2. Let `b` be the bucket size, defined as 4.
+3. Let `allocated` be the closest power of 2 that is larger than `N`.
+4. Let `bytes` be `f`\*`allocated`\*`b`/8 rounded up to the nearest integer
+5. Add 5 to `bytes`
+6. Allocate memory of `bytes` and set it to zero. Assign it to `digest-value`.
+7. Set the first byte to `P`
+8. Set the second till fifth bytes to `N` in big endian form
+9. Return the `digest-value`.
+
+Note: `allocated` is necessary due to the nature of the way Cuckoo filters are creating the
+secondary hash, by XORing the initial hash and the fingerprint's hash. The XOR operation means
+that secondary hash can pick an entry beyond the initial number of entries, up to the next power
+of 2. In order to avoid issues there, we allocate the table appropriately. For increased space
+efficiency, it is recommended that implementations pick a number of entries that's close to the
+next power of 2.
+
+### Adding a URL to the Digest-Value {#adding}
+
+Given the following inputs:
+
+* `URL` a string corresponding to the Effective Request URI ({{RFC7230}}, Section 5.5) of a cached
+response {{RFC7234}}
+* `ETag` a string corresponding to the entity-tag {{RFC7232}} of a cached response {{RFC7234}} (if
+the ETag is available; otherwise, null);
+* `maxcount` - max number of cuckoo hops
+* `digest-value`
+
+1. Let `f` be the value of the first byte of `digest-value`.
+2. Let `b` be the bucket size, defined as 4.
+3. Let `N` be the value of the second to fifth bytes of `digest-value` in big endian form.
+4. Let `key` be the return value of {{key}} with `URL` and `ETag` as inputs.
+5. Let `h1` be the return value of {{hash}} with `key` and `N` as inputs.
+6. Let `dest_fingerprint` be the return value of {{fingerprint}} with `key` and `f` as inputs.
+7. Let `h2` be the return value of {{hash2}} with `h1`, `dest_fingerprint` and `N` as inputs.
+8. Let `h` be either `h1` or `h2`, picked in random.
+9. While `maxcount` is larger than zero:
+    1. Let `position_start` be 40 + `h` * `f` \* `b`.
+    2. Let `position_end` be `position_start` + `f` \* `b`.
+    3. While `position_start` < `position_end`:
+        1. Let `bits` be `f` bits from `digest_value` starting at `position_start`.
+        2. If `bits` is all zeros, set `bits` to `dest_fingerprint` and terminate these steps.
+        3. Add `f` to `position_start`.
+    4. Let `e` be a random number from 0 to `b`.
+    5. Substract `f` * (`b` - `e`) from `position_start`.
+    6. Let `bits` be `f` bits from `digest_value` starting at `position_start`.
+    7. Let `fingerprint` be the value of bits, read as big endian.
+    8. Set `bits` to `dest_fingerprint`.
+    9. Set `dest_fingerprint` to `fingerprint`.
+    10. Let `h` be {{hash2}} with `h`, `dest_fingerprint` and `N` as inputs.
+    11. Substract 1 from `maxcount`.
+13. Substract `f` from `position_start`.
```

Comment:
Double check that the generated text and HTML are correct. In the past, list numbers higher than 9 needed to wrap around to 0 to work correctly; not sure if that's been fixed or not.

Reply:
Thank you for pointing that out.

The numbers were correctly handled by the parser, however the numbers in the markdown were wrong.

The issue has been fixed in 8469d48.